datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -608,15 +608,12 @@ impl Stream for GroupedHashAggregateStream {
         loop {
             match &self.exec_state {
                 ExecutionState::ReadingInput => 'reading_input: {
-                    match ready!(self.input.poll_next_unpin(cx)) {
-                        // new batch to aggregate
-                        Some(Ok(batch)) => {
+                    match (ready!(self.input.poll_next_unpin(cx)), self.mode) {
+                        // New batch to aggregate in partial aggregation operator
+                        (Some(Ok(batch)), AggregateMode::Partial) => {
```

Comment:
BTW another way to express this rather than matching on `(.., self.mode)` would be to use a match clause

like

```suggestion
                        Some(Ok(batch)) if self.mode == AggregateMode::Partial => {
```

Reply:
👍 Seems nice, fixed.